### PR TITLE
Added Largest Sum Contiguous SubArray [C]

### LINKED ISSUE #111: Largest Sum Contiguous Subarray [C]
[http://www.geeksforgeeks.org/largest-sum-contiguous-subarray/](http://www.geeksforgeeks.org/largest-sum-contiguous-subarray/)

### PR DESCRIPTION
**Pull request by Lucifer24**
Fixed #111 

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+int LargestSumContinousSubArray(int a[], int size) {
+	int max_till = 0;
+	int max_int = 0;
+	for (int i = 0; i < size; i++) {
+		max_int = max_int +a[i];
```
Comment (aviaryan) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Make your code style uniform. Example - there should be a space after `+`. 
```C
max_int = max_int + a[i];
```

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+int LargestSumContinousSubArray(int a[], int size) {
```
Comment (aviaryan) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Function names in C are not in ProperCase but camelCase. Please change the filename too.

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+			max_int = 0;
+		}
+	}
+	return max_till;
+}
+
+int main() {
+	int array [10];
+	for (int k = 0; k < 10; k++)
+	{
```
Comment (aviaryan) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Opening brace should be on the same line as "for loop" declaration (as per your style).

Comment (Monal5031) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Please follow the coding guidelines as mentioned in the contributing.md

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		array[k] = rand()%10;
```
Comment (Monal5031) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Space before and after every binary operator.

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#include<stdio.h>
+#include<stdlib.h>
```
Comment (Monal5031) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Please remove this unnecessary include.

File: largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c (hunk 2 above)
Comment (Monal5031) [inline largest_sum_contiguous_subarray/LargestSumContiguousSubarray.c]: Make it `const int arr*` variables name should make sense a little bit as far as possible.

**Comment by Lucifer24** [comment]
@aviaryan take a look at changes and add me to members list

**Comment by aviaryan** [comment]
Well, we have a problem here. Another PR #102 has also been created for the same thing. 
The sad part is that the language was defined as `cpp` there. 

CONTRIBUTING.md says:
> In some cases, C and C++ implementation will be similar. In that case, only the C implementation must be done.

Now in this case, we will have to close one of the 2 PRs. As Largest continuous sum is a problem whose CPP implementation is same as C, I think #102 is more in the wrong here even though it was created earlier. So #102 should be closed. 
@singhpratyush Views? 
/cc @mohitkyadav @yashLadha @Monal5031

**Comment by mohitkyadav** [comment]
Should I close #102 ?

File: largest_sum_contiguous_subarray/largestSumContiguousSubarray.c [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#include<stdio.h>
+#include<stdlib.h>
```
Comment (aviaryan) [inline largest_sum_contiguous_subarray/largestSumContiguousSubarray.c]: Space after #include and <lib>

File: largest_sum_contiguous_subarray/largestSumContiguousSubarray.c [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+		}
+		if (max_int < 0) {
+			max_int = 0;
+		}
+	}
+	return max_till;
+}
+
+int main() {
+	int array [10];
```
Comment (aviaryan) [inline largest_sum_contiguous_subarray/largestSumContiguousSubarray.c]: Make it `array[10]`, remove space.

**Comment by Monal5031** [comment]
@Lucifer24 Please resolve the conflicts.

**Comment by Lucifer24** [comment]
@Monal5031 @aviaryan take a look now

**Comment by aashutoshrathi** [comment]
@Monal5031 @aviaryan isn't stdlib needed for rand() function ?

**Comment by aviaryan** [comment]
@aashutoshrathi Oh yes. Thanks for pointing out.

**Pull request merged**